### PR TITLE
Extract the message determination code from _build_id_messages

### PR DESCRIPTION
**Pull request by jasonleinbach-wf**
## Changes
- Extracted message determination code from the _build_id_messages function
- Added tests

Please review: @hefischer @andrewperkins-wf

### CODE REVIEW

File: utilities/generate_messages_catalog.py [hunk 1]
```diff
@@ -131,96 +134,183 @@ def _is_callable(item):
...
+                            'reference_filename': ref_module_name,
+                            'line_number': item.lineno
+                        }
+                    )
+            except (AttributeError, IndexError):
+                pass
+    return id_messages
+
+
+def _get_validation_message(msgArg):
```
Comment (chrislococo-wf) [inline utilities/generate_messages_catalog.py]: CR: trying to break this habit in the arelle codebase, variables should be snake_case'd

File: test/utilities/test_generate_messages_catalog.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        find_modules.assert_has_calls(call_list, any_order=True)
+        self.assertEqual(
+            len(call_list), find_modules.call_count,
+            "Incorrect number of calls to _find_modules_and_directories"
+        )
+
+    def test_get_validation_message_str(self):
+        """Checks that _get_validation_message works when msgArg is an ast.Str"""
+        msgArg = mock.Mock(spec=ast.Str)
+        msgArg.s = "page"
```
Comment (chrislococo-wf) [inline test/utilities/test_generate_messages_catalog.py]: SG: Same jazz on all the msgArgs in the test files, please use snake_case

**Comment by chrislococo-wf** [comment]
+1

**Comment by derekgengenbacher-wf** [comment]
+1

**Comment by chrislococo-wf** [comment]
### QA Performed:
- Generated new message catalog cleanly. 
- Unit tests are clean.

### Result: +1 QA

**Pull request merged**